File: app/src/main/java/com/grayraven/mqtt_test2/MqttEventCallback.java
package com.grayraven.mqtt_test2;

import android.annotation.SuppressLint;
import android.util.Log;

import org.eclipse.paho.client.mqttv3.MqttCallback;
import org.eclipse.paho.client.mqttv3.MqttMessage;
import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;

public class MqttEventCallback implements MqttCallback {

    final String TAG = "MqttEventCb";

    @Override
    public void connectionLost(Throwable arg0) {
        Log.d(TAG, "connection lost");

    }

    @Override
    public void deliveryComplete(IMqttDeliveryToken arg0) {

        Log.i(TAG, "deliveryComplete" );
    }

    @Override
    @SuppressLint("NewApi")
    public void messageArrived(String topic, final MqttMessage msg) throws Exception {
        //  Log.i(TAG, "Message arrived from topic" + topic);

    }
}

File: app/src/main/java/com/grayraven/mqtt_test2/MqttEventCallback.kt
package com.grayraven.mqtt_test2

import android.annotation.SuppressLint
import android.util.Log
import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken
import org.eclipse.paho.client.mqttv3.MqttCallback
import org.eclipse.paho.client.mqttv3.MqttMessage

class MqttEventCallback : MqttCallback {

    internal val TAG = "MqttEventCb"

    override fun connectionLost(arg0: Throwable) {
        Log.d(TAG, "connection lost")

    }

    override fun deliveryComplete(arg0: IMqttDeliveryToken) {

        Log.i(TAG, "deliveryComplete")
    }

    @SuppressLint("NewApi")
    @Throws(Exception::class)
    override fun messageArrived(topic: String, msg: MqttMessage) {
          Log.i(TAG, "Message arrived from topic" + topic);

    }
}
